### PR TITLE
github: update CI

### PR DESCRIPTION
**Pull request by RaduBerinde**
- update CI to run with go 1.22 and 1.23
 - require go 1.22 in go.mod
 - run `go mod tidy`
 - change Makefiles to use `go install` instead of `go get` (which
   messes with go.mod)
 - fix a lint failure

### CODE REVIEW

**Comment by RaduBerinde** [comment]
CI is still flaky because of https://github.com/golang/go/issues/70305 causes a failure in a randomized test.

**Comment by RaduBerinde** [comment]
Removed the offending `GoString` tests.

File: Makefile [hunk 1]
```diff
@@ -150,20 +150,20 @@ tests:
 	(cd test/stdtypes && make test)
 
 vet:
-	go get golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow
+	go install golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow@latest
```
Comment (tbg) [inline Makefile]: let's pin this to a specific version.

File: Makefile [hunk 2]
```diff
@@ -150,20 +150,20 @@ tests:
...
 	go vet ./...
 	go vet -vettool=$(shell which shadow) ./...
 
 errcheck:
-	go get github.com/kisielk/errcheck
+	go install github.com/kisielk/errcheck@latest
```
Comment (tbg) [inline Makefile]: let's pin this to a specific version.

File: Makefile [hunk 3]
```diff
@@ -150,20 +150,20 @@ tests:
...
 	errcheck ./test/...
 
 drone:
 	sudo apt-get install protobuf-compiler
 	(cd $(GOPATH)/src/github.com/gogo/protobuf && make buildserverall)
 
 testall:
-	go get -u github.com/golang/protobuf/proto
+	go install github.com/golang/protobuf/proto@latest
```
Comment (tbg) [inline Makefile]: let's pin this to a specific version.

File: Makefile [hunk 4]
```diff
@@ -172,7 +172,7 @@ testall:
 	make tests
 
 bench:
-	go get golang.org/x/tools/cmd/benchcmp
+	go install golang.org/x/tools/cmd/benchcmp@latest
```
Comment (tbg) [inline Makefile]: let's pin this to a specific version.

**Review by tbg — APPROVED** [review]
Thank you for doing this! LGTM but please pin all the dependencies :) I don't like surprises.

File: Makefile (hunk 3 above)
Comment (RaduBerinde) [reply Makefile]: Reverted this, it didn't work since it's not a binary. It's not being called in CI.

**Comment by RaduBerinde** [comment]
TFTR!

**Pull request merged**